src/components/quantity-input/quantity-input.tsx

```diff
@@ -4,60 +4,56 @@ import styles from './quantity-input.module.scss';
 import { useRef } from 'react';
 
 type QuantityInputProps = {
-    value: number | null;
-    onChange: (value: number | null) => void;
+    value: number;
+    onChange: (value: number) => void;
     id?: string;
     className?: string;
 };
 
 export const QuantityInput = ({ value, onChange, id, className }: QuantityInputProps) => {
     const lastValidQuantity = useRef(value);
 
-    const handleChange = (newValue: number | null) => {
+    const handleChange = (newValue: number) => {
         onChange(newValue);
-        if (newValue !== null) {
+        if (newValue !== 0) {
             lastValidQuantity.current = newValue;
         }
     };
 
     const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
         const value = event.target.value;
-        if (!value) handleChange(null);
+        if (!value) handleChange(0);
         const number = Number.parseInt(value, 10);
         if (!Number.isNaN(number) && number > 0) {
             handleChange(number);
         }
     };
 
     const decrease = () => {
-        if (value !== null && value > 1) {
+        if (value > 1) {
             handleChange(value - 1);
         }
     };
 
     const increase = () => {
-        handleChange(value === null ? 1 : value + 1);
+        handleChange(value + 1);
     };
 
     const handleInputBlur = () => {
-        if (value === null && lastValidQuantity.current !== null) {
+        if (value === 0 && lastValidQuantity.current) {
             handleChange(lastValidQuantity.current);
         }
     };
 
     return (
         <div className={styles.root}>
-            <button
-                className={styles.button}
-                onClick={decrease}
-                disabled={value === null || value === 1}
-            >
+            <button className={styles.button} onClick={decrease} disabled={value <= 1}>
                 <MinusIcon className={styles.icon} />
             </button>
             <input
                 id={id}
                 className={classNames(styles.input, className)}
-                value={value ?? ''}
+                value={value || ''}
```

Comment:
What is `|| ''` for? If someone is passing zero, wouldn't it make more sense to show zero?